homeassistant/components/geniushub/__init__.py

```diff
@@ -202,11 +212,14 @@ def device_state_attributes(self) -> Dict[str, Any]:
 class GeniusZone(GeniusEntity):
     """Base for all Genius Hub zones."""
 
-    def __init__(self):
+    def __init__(self, zone) -> None:
         """Initialize the Zone."""
         super().__init__()
 
-        self._zone = None
+        self._zone = zone
+        # pylint: disable=protected-access
+        self._unique_id = f"{zone._hub.uid}_zone_{zone.id}" if zone._hub.uid else None
```

Comment:
I can have `unique_id`s for v3 api users only, or I can asked v1 api users to add teh Hub's UID to configuration.yaml (it's the Hub's MAC address) - a breaking change:
```python
_V1_API_SCHEMA = vol.Schema(
    {vol.Required(CONF_TOKEN): cv.string, vol.Required(CONF_UID): cv.string}
)

...

uid = device._hub.uid if device._hub.uid else self.hass.data[DOMAIN][CONF_UID]
self._unique_id = f"{uid}_device_{device.id}"

```

Reply:
Can we check the mac address of the hub with a tool like `getmac`?
https://pypi.org/project/getmac/

Reply:
Well, I can't get it to work in my testbed, as it relies upon **arp** (because NAT), but I cannot see why it wouldn't work... _If_ HASS & Hub are on the same subnet.

However, the user will have to explicitly put the Hub's IP-address in their `configuration.yaml` as there is no way to discover the Hub, short of a port scanner... and, if the IP address is configured manually, and the Hub's address subsequently changes, then we're in trouble.

My thinking is that, if we're going to ask user to put an IP address in `configuration.yaml`, they may as well pick up the back of the hub and read off the MAC address, and use that instead.

I think this is the way to go:
```
_V1_API_SCHEMA = vol.Schema(
    {vol.Required(CONF_TOKEN): cv.string, vol.Required(CONF_UID): cv.string}
)
```

Reply:
Yes, we shouldn't ask for an ip address if that isn't required for the connection to the hub.

It's ok to ask for a hub id.